src/app/view/components/dot-navigation/dot-navigation.component.ts

```diff
@@ -25,21 +44,55 @@ export class DotNavigationComponent implements OnInit {
      * @param {string} id menu item id
      * @memberof MainNavigationComponent
      */
-    onClick(event: MouseEvent, id: string): void {
+    onClick($event: {originalEvent: MouseEvent, data: DotMenuItem}): void {
         event.stopPropagation();
-        if (!event.ctrlKey && !event.metaKey) {
-            this.dotNavigationService.reloadCurrentPortlet(id);
+
+        if (!$event.originalEvent.ctrlKey && !$event.originalEvent.metaKey) {
+            this.dotNavigationService.reloadCurrentPortlet($event.data.id);
         }
     }
 
     /**
-     * Check is the portlet in the nav is active.
+     * Set isOpen to the passed DotMenu item
      *
-     * @param {string} id
-     * @returns
+     * @param {DotMenu} currentItem
      * @memberof DotNavigationComponent
      */
-    isActive(id: string) {
-        return this.dotNavigationService.isActive(id);
+    onMenuClick(event: {originalEvent: MouseEvent, data: DotMenu}): void {
+        this.change.emit();
+
+        if (this.collapsed) {
+            this.dotNavigationService.goTo(event.data.menuItems[0].menuLink);
+        }
+
+        this.menu = this.menu.map((item: DotMenu) => {
+            if (item.isOpen) {
+                item.isOpen = false;
+            } else {
+                item.isOpen = event.data.id === item.id;
+            }
+            return item;
+        });
+    }
+
+    private setActive(id: string) {
+        this.menu = this.menu.map((item: DotMenu) => this.isMenuActive(item, id));
+    }
+
+    private isMenuActive(menu: DotMenu, id: string): DotMenu {
+        let isActive = false;
+
+        menu.menuItems.forEach((item: DotMenuItem) => {
+            if (item.id === id) {
+                item.active = true;
```

Comment:
we should take this set statement out of here or change the name of the method, if the method name is 'isMenuActive' I expect that this just return a boolean and it don't set anything